libsolidity/interface/CompilerStack.cpp

```diff
@@ -150,10 +149,12 @@ bool CompilerStack::parse()
 			}
 		}
 	}
-	if (!Error::containsOnlyWarnings(m_errors))
-		// errors while parsing. should stop before type checking
-		return false;
 
+	return Error::containsOnlyWarnings(m_errors);
```

Comment:
do you mean like this?:

if (!Error::containsOnlyWarnings(m_errors))
	{
		m_errors.clear();
		return false;
	}
	else
		return true;`
`

Does it make sense to clean m_errors when they occur? How would you then know what the errors were?